### PR TITLE
pythonPackages.(datasette, mergedeep, asgi-csrf, starlette, httpx, httpcore): upgrades & new additions

### PR DESCRIPTION
**Pull request by risicle**
###### Motivation for this change
This started with me upgrading `datasette` from 0.39 -> 0.46, in an attempt to mitigate https://github.com/simonw/datasette/security/advisories/GHSA-q6j3-c4wc-63vw.

This required the addition of `mergedeep` and `asgi-csrf`.

The addition of `asgi-csrf` required a bump of `starlette` and `httpx` to get the tests to pass.

The bump of `httpx` required the addition of `httpcore`.

It's been a fun few evenings.

In the bumped packages I also performed a few cleanups, looking for no-longer-necessary dependencies, removing outdated comments etc.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
@ofborg build python37Packages.datasette python37Packages.fastapi google-music-scripts

**Comment by risicle** [comment]
Hmm is the aarch64 failure just `python38Packages.websockets` being broken...?

**Comment by wd15** [comment]
Builds for me.

**Review by wd15 — APPROVED** [review]

**Comment by jonringer** [comment]
sorry, I sorted python-packages, please rebase

**Review by wd15 — APPROVED** [review]

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -3587,6 +3591,8 @@ in {
 
   merkletools = callPackage ../development/python-modules/merkletools { };
 
+  mergedeep = callPackage ../development/python-modules/mergedeep { };
+
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:3595]: should be switched
```suggestion
  mergedeep = callPackage ../development/python-modules/mergedeep { };

  merkletools = callPackage ../development/python-modules/merkletools { };

```

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -379,6 +379,8 @@ in {
 
   asgiref = callPackage ../development/python-modules/asgiref { };
 
+  asgi-csrf = callPackage ../development/python-modules/asgi-csrf { };
+
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:383]:
```suggestion
  asgi-csrf = callPackage ../development/python-modules/asgi-csrf { };

  asgiref = callPackage ../development/python-modules/asgiref { };

```

**Review by jonringer — APPROVED** [review]
diff LGTM

```
https://github.com/NixOS/nixpkgs/pull/96565
17 packages built:
datasette google-music-scripts python37Packages.asgi-csrf python37Packages.datasette python37Packages.fastapi python37Packages.google-music python37Packages.httpcore python37Packages.httpx python37Packages.mergedeep python37Packages.starlette python38Packages.asgi-csrf python38Packages.fastapi python38Packages.google-music python38Packages.httpcore python38Packages.httpx python38Packages.mergedeep python38Packages.starlette
```

**Pull request merged**